infra/index.ts

```diff
@@ -0,0 +1,37 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+import * as awsx from "@pulumi/awsx";
+
+import exampleHandler from "../src/lambda";
+
+type Stage = "dev" | "beta" | "prod";
+
+// Read the stage from the STAGE environment variable
+const STAGE = process.env.STAGE as Stage ?? "dev";
+const SERVICE_NAME = "service-template";
+
+const createResourceName = (name: string) => `${STAGE}-${SERVICE_NAME}-${name}`;
```

Comment:
think it'd be nicer to have a slash as a separator for the resource name if there's already dashes being used with the names
 
```suggestion
const createResourceName = (name: string) => `${STAGE}/${SERVICE_NAME}/${name}`;
```

Reply:
We spoke offline and we'll go with stage_service-name_lambda-name for Lambdas, and stage-service-name-bucket-name for S3 buckets